src/python_testing/TC_IDM_2_2.py

```diff
@@ -0,0 +1,596 @@
+        self.print_step(22, "Send the Read Request Message to the DUT 3 times and check if they are correct each time")
+        read_request_1 = await self.default_controller.ReadAttribute(self.dut_node_id, [Clusters.Objects.Descriptor.Attributes.ServerList])
+        read_request_2 = await self.default_controller.ReadAttribute(self.dut_node_id, [Clusters.Objects.Descriptor.Attributes.ServerList])
+        read_request_3 = await self.default_controller.ReadAttribute(self.dut_node_id, [Clusters.Objects.Descriptor.Attributes.ServerList])
+        asserts.assert_equal(read_request_1, read_request_2)
+        asserts.assert_equal(read_request_2, read_request_3)
+
+        # Step 23
+        # TH sends a Read Request Message to the DUT to read a particular attribute with the DataVersionFilter Field not set.
+        # DUT sends back the attribute value with the DataVersion of the cluster.
+        # TH sends a second read request to the same cluster with the DataVersionFilter Field set with the dataversion value received before.
+        # DUT should not send a report data action with the attribute value to the TH if the data version is same as that requested.
+        self.print_step(
+            23, "Send the Read Request Message to the DUT to read a particular attribute with the DataVersionFilter Field not set")
+        read_request = await self.default_controller.ReadAttribute(self.dut_node_id, [(0, Clusters.Objects.Descriptor.Attributes.ServerList)])
+        data_version = read_request[0][Clusters.Descriptor][Clusters.Attribute.DataVersion]
+        data_version_filter = [(0, Clusters.Descriptor, data_version)]
+
+        read_request_2 = await self.default_controller.ReadAttribute(self.dut_node_id, [(0, Clusters.Objects.Descriptor.Attributes.ServerList)], dataVersionFilters=data_version_filter)
+        asserts.assert_equal(read_request_2, {})
+
+        # Step 24
+        # TH sends a Read Request Message to the DUT to read a particular attribute with the DataVersionFilter Field not set.
+        # DUT sends back the attribute value with the DataVersion of the cluster.
+        # TH sends a write request to the same cluster to write to any attribute.
+        # TH sends a second read request to read an attribute from the same cluster with the DataVersionFilter Field set with the dataversion value received before.
+        # DUT should send a report data action with the attribute value to the TH.
+
+        self.print_step(
+            24, "Send the Read Request Message to the DUT to read a particular attribute with the DataVersionFilter Field not set")
+
+        read_request = await self.default_controller.ReadAttribute(self.dut_node_id, [(0, Clusters.Objects.BasicInformation.Attributes.NodeLabel)])
+        data_version_1 = read_request[0][Clusters.BasicInformation][Clusters.Attribute.DataVersion]
+        data_version_value = 123456
```

Comment:
This is the node label value, right? Can you change the variable name to reflect that?

also, can this be a random string? node label isn't a number

Reply:
This caused a type exception when I changed this. I added the change, but it is commented out for now.

Reply:
It's definitely a string, so if you're getting an exception, that indicates there is a bug somewhere that needs to be tracked down. 

Ex. see https://github.com/project-chip/connectedhomeip/blob/master/src/python_testing/TC_IDM_4_2.py#L536

Reply:
This turned out to be a bug in an assert statement in my code (when the misassumption was that it was supposed to be an integer, it used int() to turn it into a string). This is now fixed.